### ISSUE TITLE
How to decode a struct of arbitrary fields?

### ISSUE THREAD

**Issue by ahoward1024** [alexejk/go-xmlrpc#77]
Hi, I'm struggling to decode an XMLRPC example where there are is a struct with a number of members whose names are arbitrary, so I can't put them into a tag.

<details>
<summary>Example XML</summary>

```
<methodResponse>
  <params>
    <param>
      <value>
        <struct>
          <member>
            <name>TESTING1</name>
            <value>
              <array>
                <data>
                  <value>
                    <array>
                      <data>
                        <value>
                          <struct>
                            <member><name>id</name><value><string>1009470</string></value></member>
                            <member><name>pub_date</name><value><string>2020-01-11 00:00:00</string></value></member>
                            <member><name>title</name><value><string>TITLE</string></value></member>
                        </struct></value>
                        <value><struct>
                            <member><name>title</name><value><string>TITLE2</string></value></member>
                            <member><name>pub_date</name><value><string>2020-01-11 00:00:00</string></value></member>
                            <member><name>id</name><value><string>1009879</string></value></member>
                        </struct></value>
                        <value><struct>
                            <member><name>title</name><value><string>Title3</string></value></member>
                            <member><name>pub_date</name><value><string>2020-01-13 17:16:49</string></value></member>
                            <member><name>id</name><value><string>1304451</string></value></member>
                        </struct></value>
                  </data></array></value>
            </data></array></value>
          </member>
          <member>
            <name>TESTING2</name>
            <value>
              <array>
                <data>
                  <value>
                    <array>
                      <data>
                        <value>
                          <struct>
                            <member><name>id</name><value><string>1329812</string></value></member>
                            <member><name>pub_date</name><value><string>2020-01-11 00:00:00</string></value></member>
                            <member><name>title</name><value><string>NewTitle</string></value></member>
                        </struct></value>
                        <value><struct>
                            <member><name>title</name><value><string>NextTitle</string></value></member>
                            <member><name>pub_date</name><value><string>2021-01-11 00:00:00</string></value></member>
                            <member><name>id</name><value><string>1489372</string></value></member>
                        </struct></value>
                        <value><struct>
                            <member><name>title</name><value><string>Title12</string></value></member>
                            <member><name>pub_date</name><value><string>2020-01-13 17:16:49</string></value></member>
                            <member><name>id</name><value><string>1229276</string></value></member>
                        </struct></value>
                  </data></array></value>
            </data></array></value>
          </member>
        </struct>
      </value>
    </param>
  </params>
</methodResponse>

```

</details>

The following structure can be successfully decoded into:
```
type DataType struct {
	Id      string `json:"id" xmlrpc:"id"`
	PubDate string `json:"pub_date" xmlrpc:"pub_date"`
	Title   string `json:"title" xmlrpc:"title"`
}

type Data struct {
	Struct struct {
		TESTING1 [][]DataType
		TESTING2 [][]DataType
	}
}
```

However, **I _cannot_ know the names where `TESTING1` and `TESTING2` are**. They will change at runtime and the dataset will get added to and removed from, so those values are essentially arbitrary and therefore cannot be written directly into the struct. I also don't have control over the RPC API, so I cannot change the structure.

Any help would be greatly appreciated.

**Comment by alexejk** [comment]
Hi @ahoward1024 ,

Thanks for reaching out regarding this. I think it's a fairly new use-case I haven't heard of before. Typically when working with XML-RPC you have a knowledge of the field values but what you need is handling a struct as a map.

I took a stab at prototyping how this could be achieved and will likely open a PR fairly soon, however I'm still thinking about how to handle this in the best way from the compatibility perspective.

**Comment by alexejk** [comment]
@ahoward1024 I've pushed the PR for this fix. 
Could you please give it a try by pulling the dev branch into your go.mod?

You can do this by running 
`go get -u alexejk.io/go-xmlrpc@decode-struct-as-map`

It should give you a pseudo-version dependency entry, something like:
`alexejk.io/go-xmlrpc v0.4.2-0.20240214200317-abbf585c7d64`

In order to solve your issue you should be able to now do something like this:
```
type Data struct {
	Map map[string][][]DataType
}
```

You can check out the added tests for more details.

**Comment by ahoward1024** [comment]
@alexejk Apologies, I've been traveling and just got back to this.

I've pulled in your changes and they work flawlessly for our purpose. Thank you for getting back to me so quickly on this! :rocket:

I'll go ahead an close this out!

**Comment by alexejk** [comment]
Thanks for confirming, @ahoward1024 !
I've merged the PR and just triggered the 0.5.0 release. You can swap over to that version shortly (effectively just let it pull the latest version instead).
